2.4/blocks/aspirelists/block_aspirelists.php

```diff
@@ -3,44 +3,93 @@
-	global $COURSE;
+    function get_content() {
+        global $CFG;
+        global $COURSE;
 
-    if ($this->content !== NULL) {
-      return $this->content;
-    }
+        if ($this->content !== NULL) {
+            return $this->content;
+        }
+
+        $hrefTarget = get_config('aspirelists', 'openNewWindow');
+        $target ='_self';
+        if($hrefTarget == 1){
+            $target = '_blank';
+        }
+
+        $this->content =    new stdClass;
+
+        if ($COURSE->idnumber)
+        {
+            $cache = cache::make('block_aspirelists', 'aspirelists');
+            $lists = $cache->get('list-' . $COURSE->idnumber);
+
+            if(!$lists) {
+                // get the code from the global course object
+                $lists = $this->getTalisAspireList($COURSE->idnumber);
+                $cache->set('list-' . $COURSE->idnumber, $lists);
+            }
+
+            $output = '';
+
+            foreach ($lists as $list)
+            {
+                $itemNoun = ($list['count'] == 1) ? get_string("item", 'block_aspirelists') : get_string("items", 'block_aspirelists'); // get a friendly, human readable noun for the items
+
+                // finally, we're ready to output information to the browser
 
-	$site = get_config('aspirelists', 'targetAspire');
-    $httpsAlias = get_config('aspirelists', 'targetAspireAlias');
-
-	if (empty($site))
-	{
-		$this->content->text = get_string('no_base_url_configured', 'block_aspirelists');
-		return $this->content;
-	}
-
-	$targetKG = get_config('aspirelists', 'targetKG');
-	if (empty($targetKG))
-	{
-		$targetKG = "modules"; // default to modules
-	}
-
-    $hrefTarget = get_config('aspirelists', 'openNewWindow');
-    $target ='_self';
-    if($hrefTarget == 1){
-        $target = '_blank';
+                // item count display
+                $itemCountHtml = '';
+                if ($list['count'] > 0) // add the item count if there are any
+                {
+                    $itemCountHtml = html_writer::tag('span', " ({$list['count']} {$itemNoun})" ,array('class'=>'aspirelists-item-count'));
+                }
+
+                // last update display
+                $lastUpdatedHtml = '';
+                if (isset($list["lastUpdatedDate"]))
+                {
+                    $lastUpdatedHtml = html_writer::tag('span',', '.get_string('lastUpdated','block_aspirelists').' '.$this->contextualTime(strtotime($list["lastUpdatedDate"])) , array('class'=>'aspirelists-last-updated'));
+                }
+
+                // put it all together
+                $output .= html_writer::tag('p',
+                    html_writer::tag('a', $list['name'] , array('href' => $list['url'], 'target' => $target)) . html_writer::empty_tag('br') . $itemCountHtml . $lastUpdatedHtml );
+            }
+
+            if ($output=='')
+            {
+                    $this->content->text     = html_writer::tag('p', get_config('aspirelists', 'noResourceListsMessage'));
```

Comment:
There's some extra whitespace before the ``=`` on this line.  Theres a few other places this is happening the file.  It would be great to tidy these up before merging.

Reply:
I've tidied up the additional whitespace before the `=` on a number of lines.